app/src/test/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNamePresenterTest.java

```diff
@@ -0,0 +1,64 @@
+package io.intrepid.contest.screens.entrysubmission.entryname;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.Presenter;
+import io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.View;
+import io.intrepid.contest.testutils.BasePresenterTest;
+import io.intrepid.contest.testutils.TestPresenterConfiguration;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EntryNamePresenterTest extends BasePresenterTest<EntryNamePresenter> {
+    @Mock
+    View mockView;
+
+    private Presenter presenter;
+
+    @Before
+    public void setup() {
+        presenter = new EntryNamePresenter(mockView, TestPresenterConfiguration.createTestConfiguration());
+    }
+
+    @Test
+    public void contestantWelcomeTextViewShouldShowContestName() {
+        presenter.onViewCreated();
+        verify(mockView).onContestNameRetrieved("Chili Cookoff");
+    }
+
+    @Test
+    public void entryNameTextViewFocusedShouldShowNextButton() {
+        presenter.onEntryNameTextViewFocusChanged(true);
+        verify(mockView).showEntryNameNextButton();
+        verify(mockView, never()).hideEntryNameNextButton();
+    }
+
+    @Test
+    public void entryNameTextViewFocusLostShouldHideNextButton() {
+        presenter.onEntryNameTextViewFocusChanged(false);
+        verify(mockView).hideEntryNameNextButton();
+        verify(mockView, never()).showEntryNameNextButton();
+    }
+
+    @Test
+    public void entryNameTextViewEmptyShouldDisableNextButton() {
+        CharSequence newText = "";
```

Comment:
This is an awesome place to have a sidebar conversation about testing. In testing we do three things, we:
1) Arrange
2) Act
3) Assert. 

if there is only one of each, dont put spaces. If there are more than any, put spaces through each of these. To clarify, arranging is setting up things we will use in the test (line 51 here). Then we will act, which is usually only one line of what we're testing on the presenter (line 52). Then we Assert (lines 53, and 54). Asserts are usually related to mockviews or literal assertions.